game/scripts/vscripts/abilities/oaa_zuus_cloud.lua

```diff
@@ -0,0 +1,155 @@
+zuus_cloud_oaa = class( AbilityBaseClass )
+LinkLuaModifier( "modifier_zuus_cloud_oaa", "abilities/oaa_zuus_cloud.lua", LUA_MODIFIER_MOTION_NONE )
+
+function zuus_cloud_oaa:OnSpellStart()
+  local caster = self:GetCaster()
+  local hCloud = CreateUnitByName( "npc_dota_zeus_cloud", self:GetCursorPosition(), true, caster, caster, caster:GetTeamNumber() )
+  hCloud:SetOwner( self:GetCaster() )
+  hCloud:SetControllableByPlayer( self:GetCaster():GetPlayerOwnerID(), false )
+  hCloud:AddNewModifier( caster, self, "modifier_zuus_cloud_oaa", nil )
+  hCloud:AddNewModifier( caster, self, "modifier_kill", { duration = self:GetSpecialValueFor( "cloud_duration" ) } )
+  FindClearSpaceForUnit( hCloud, self:GetCursorPosition(), true )
```

Comment:
On that note of being a flying unit though, it does seem to obstruct movement. Though the vanilla one also does apparently, for some strange reason.

Reply:
Since both this and vanilla does have collision (I think it is necessary sor it baing able to be hit) I'll keep this way

Reply:
It's not. Bosses recently had collision disabled so that people wouldn't get stuck in them. Since for some reason, flying movement doesn't automatically disable pathing collision. But anyway, probably best to just keep parity with vanilla.

Reply:
waitwaitwaitwaitwaitwaitwaitwaitwait base dota nimbus cloud has collision? That's awesome. Poop wards alsohave collision but it's less awesome.